src/dialect/snowflake.rs

```diff
@@ -149,6 +153,44 @@ impl Dialect for SnowflakeDialect {
         None
     }
 
+    fn parse_column_option(
+        &self,
+        parser: &mut Parser,
+    ) -> Option<Result<Option<ColumnOption>, ParserError>> {
+        let with = parser.parse_keyword(Keyword::WITH);
+
+        if parser.parse_keyword(Keyword::IDENTITY) {
+            Some(
+                parse_identity_property(parser)
+                    .map(|p| Some(ColumnOption::Identity(IdentityPropertyKind::Identity(p)))),
+            )
+        } else if parser.parse_keyword(Keyword::AUTOINCREMENT) {
+            Some(parse_identity_property(parser).map(|p| {
+                Some(ColumnOption::Identity(IdentityPropertyKind::Autoincrement(
+                    p,
+                )))
+            }))
+        } else if parser.parse_keywords(&[Keyword::MASKING, Keyword::POLICY]) {
+            Some(
+                parse_column_policy_property(parser, with)
+                    .map(|p| Some(ColumnOption::Policy(ColumnPolicy::MaskingPolicy(p)))),
+            )
+        } else if parser.parse_keywords(&[Keyword::PROJECTION, Keyword::POLICY]) {
+            Some(
+                parse_column_policy_property(parser, with)
+                    .map(|p| Some(ColumnOption::Policy(ColumnPolicy::ProjectionPolicy(p)))),
+            )
+        } else if parser.parse_keywords(&[Keyword::TAG]) {
+            Some(parse_column_tags(parser, with).map(|p| Some(ColumnOption::Tags(p))))
+        } else {
+            // needs to revert initial state of parser if dialect finds any matching
+            if with {
+                parser.prev_token();
+            }
```

Comment:
for this part can we use `parser.maybe_parse`? I think that would handle this automatically

Reply:
Applied.